# Analysis of SQ Degradation Pathways in Human Gut Microbiome

This pipeline identifies and characterizes microbial enzymes and gene clusters involved in **sulfoquinovose (SQ) degradation** across gut microbiomes of vegetarians and non-vegetarians.

It integrates HMMER, DIAMOND, genome annotations, and pathway reconstruction to detect and visualize conserved metabolic loci.

---

## ğŸ§ª Project Highlights

- Uses **DIAMOND** and **JackHMMER** to find homologs of SQ-degrading enzymes in the **UHGP** database.
- Annotates gene hits with pathway and taxonomic information.
- Groups enzymes into **co-located loci** using GFF annotation.
- Detects **core metabolic pathways** (sulfo-EMP, -TK, -TAL, -ED).
- Provide ANCOM-BC for 16S rRNA amplicon data with **QIIME**. 
- Mapped 16S rRNA amplicon to Metagenomic data by **bwa-mem** and **coverM**.
- Performs **visualization** and **OLS statistical modeling** of pathway abundance.
  

---

## ğŸ“ Directory Structure
---
```
.
â”œâ”€â”€ code
â”‚   â”œâ”€â”€ bin
â”‚   â”‚   â”œâ”€â”€ amplicon_data_analysis.sh
â”‚   â”‚   â”œâ”€â”€ bwa_samtools.sh
â”‚   â”‚   â”œâ”€â”€ coverm.sh
â”‚   â”‚   â”œâ”€â”€ DA_species_masline2.R
â”‚   â”‚   â”œâ”€â”€ diamond.sh
â”‚   â”‚   â”œâ”€â”€ exctractlocuses.py
â”‚   â”‚   â”œâ”€â”€ ExtractLocalGroup.py
â”‚   â”‚   â”œâ”€â”€ FindLocalGroup.py
â”‚   â”‚   â”œâ”€â”€ jackhmmer.sh
â”‚   â”‚   â”œâ”€â”€ mmseq_cluster.sh
â”‚   â”‚   â””â”€â”€ phyl.R
â”‚   â””â”€â”€ SQDegradationAnalysis.ipynb
â”œâ”€â”€ data
â”‚   â”œâ”€â”€ AGPsamplemetadata.csv
â”‚   â”œâ”€â”€ LocusMetadata.csv
â”‚   â”œâ”€â”€ SQ_degradation_enzymes.fasta
â”‚   â””â”€â”€ SQDegradationEnzymesMeta.csv
â”œâ”€â”€ environment.yml
â””â”€â”€ README.md
```

### ğŸ”§ Script Descriptions

- [`amplicon_data_analysis.sh`](code/bin/amplicon_data_analysis.sh) â€” Bash script for processing amplicon variation data.
- [`bwa_samtools.sh`](code/bin/bwa_samtools.sh) â€” Bash pipeline for mapping reads using BWA and processing alignments with SAMtools.
- [`coverm.sh`](code/bin/coverm.sh) â€” Calculates RPKM values from read mappings for AGP locus coverage analysis (bash).
- [`DA_species_masline2.R`](code/bin/DA_species_masline2.R) â€” Builds a MASLiNE model for differential abundance of species (R).
- [`diamond.sh`](code/bin/diamond.sh) â€” Performs DIAMOND-based protein alignment (bash).
- [`exctractlocuses.py`](code/bin/exctractlocuses.py) â€” Extracts gene clusters with Â±1000 bp flanks into FASTA format (adjustable).
- [`ExtractLocalGroup.py`](code/bin/ExtractLocalGroup.py) â€” Aggregates gene cluster DataFrames from `FindLocalGroup` output files into a single table (python).
- [`FindLocalGroup.py`](code/bin/FindLocalGroup.py) â€” Detects co-localized genes within a 5000 bp window and generates annotated GFF-style DataFrames (python).
- [`jackhmmer.sh`](code/bin/jackhmmer.sh) â€” Runs JackHMMER protein alignments against the target database (bash).
- [`phyl.R`](code/bin/phyl.R) â€” Infers phylogenetic relationships based on the identified gene clusters (R).
- [`mmseq_cluster.sh`](code/bin/mmseq_cluster.sh) â€” Cluster gene locus with different settings.


### ğŸ—ƒï¸ Data Description

- [AGPsamplesmetadata](data/AGPsamplemetadata.csv) - Metadata information for selected AGP samples
- [AGPsamplesmetadata](data/AGPsamplemetadata.csv) - Metadata information for selected AGP samples
- [LocusMetadata](data/LocusMetadata.csv) - Information about gene clusters
- [SQ_degradation_enzymes](data/SQ_degradation_enzymes.fasta) - Fasta file with proteins linked to SQ-degradation pathways
- [SQD_locus_for_mapping](data/SQD_locus_for_mapping.fasta) - Fasta file with retieved SQ-degradation locuses from UHGG samples
- [SQDegradation](data/SQDegradationEnzymesMeta.csv) - Metadata infomation for SQ-degradation locuses
- [SQDegradationEnzymesMeta](data/SQDegradationEnzymesMeta.csv) - Metadata infomation for SQ-degradation proteinsyour_email

---

## âš™ï¸ Requirements

You can create the environment with:

```bash
conda env create -f environment.yml
conda activate sqd-analysis
```

## ğŸ“¬ Contact

For questions, suggestions, or workflow errors feel free to open an issue or contact [anurovartemiy@gmail.com].

